### PR TITLE
MHV-57141 sm update import patient keyword search page

### PR DESCRIPTION
**Pull request by fazilqa**
## Summary

- E2E tests for Inbox page filter-sort by keyboard required
- This PR adds new spec with required tests, additional methods and parametrization
- VA.gov - Secure Messaging

## Related issue(s)

- https://jira.devops.va.gov/browse/MHV-57141

## Testing done

- Test run locally based on 200x pattern
- All specs run in headless mode with no fails

## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  |        |       |
| Desktop |        |       |

## What areas of the site does it impact?

*(Describe what parts of the site are impacted **if** code touched other areas)*

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/mhv-secure-messaging/main/tests/e2e/keyboard-nav-test/secure-messaging-keyboard-nav-inbox-filter-sort.cypress.spec.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  };
+
+  beforeEach(() => {
+    site.login();
+    landingPage.loadInboxMessages();
+  });
+
+  it('Verify filter works correctly', () => {
+    landingPage.inputFilterDataByKeyboard('test');
+    landingPage.submitFilerByKeyboard(filteredData);
```
Comment (oleksii-morgun) [inline src/applications/mhv-secure-messaging/main/tests/e2e/keyboard-nav-test/secure-messaging-keyboard-nav-inbox-filter-sort.cypress.spec.js:22]:
```suggestion
    landingPage.submitFilterByKeyboard(filteredData);
```

File: src/applications/mhv-secure-messaging/main/tests/e2e/pages/PatientInboxPage.js [hunk 2]
```diff
@@ -523,7 +501,37 @@ class PatientInboxPage {
...
+    cy.get(Locators.CLEAR_FILTERS).click({ force: true });
+  };
+
+  inputFilterDataByKeyboard = text => {
+    cy.tabToElement('#inputField')
+      .first()
+      .type(`${text}`, { force: true });
+  };
+
+  submitFilerByKeyboard = mockFilterResponse => {
```
Comment (oleksii-morgun) [inline src/applications/mhv-secure-messaging/main/tests/e2e/pages/PatientInboxPage.js:514]: same as above
```suggestion
  submitFilterByKeyboard = mockFilterResponse => {
```

**Review by alex-morgun-1 — DISMISSED** [review]
Hey @fazilqa I will need to place this PR on a brief hold before we can proceed. This is related to another minor folder structure change upcoming. I will post more details in our engineering channel

**Review by powellkerry — APPROVED** [review]

**Comment by fazilqa** [comment]
closed

**Pull request closed without merging**